neurons/validator.py

```diff
@@ -120,6 +120,49 @@ def forward(self, job: Job) -> dict:
     def get_pdbs_to_exclude(self) -> List[str]:
         # Set of pdbs that are currently in the process of running + old submitted simulations.
         return list(self.store._db.index)
+    
+    def sample_random_uids(self, num_uids_to_sample:int, exclude_uids:List[int]) -> List[int]:
+        """ Helper function to sample a batch of uids on the network, 
+        determine their serving status, and sample more until a desired 
+        number of uids is found.
+
+        Recurrsion is used to sample until there is enough uids. 
+
+        Args:
+            exclude_uids (List[int]): a list of uids that we know we shouldn't use.
+        """
+        valid_uids = []
+        reported_compute = []
+
+        sampled_uids:torch.Tensor = get_random_uids(
+            self, k = num_uids_to_sample, exclude=exclude_uids
+        )
+
+        #TODO: Should this be async? 
+        ping_report = run_ping_step(self, uids = sampled_uids, timeout=self.config.neuron.ping_timeout)
+        can_serve = ping_report['miner_status'] #list of booleans
+
+        valid_uids.extend(np.array(sampled_uids)[can_serve].tolist())
+        reported_compute.extend(np.array(ping_report['reported_compute'])[can_serve].tolist())
+
+        deficit = num_uids_to_sample - sum(can_serve)
+
+        # If there are not enough uids, you will obtain the remainder
+        # And since you have already added the valid uids to the growing list, we can just return the valid_uids 
+        if len(sampled_uids) < num_uids_to_sample:
+            return valid_uids
+
+        if deficit > 0: 
+            exclude_uids.extend(sampled_uids)  #exclude all uids sampled.
+
+            #recursive call adding to the valid uids
+            valid_uids.extend(self.sample_random_uids(
+                num_uids_to_sample = deficit, 
+                exclude_uids=exclude_uids)
+            ) 
+
+        return valid_uids
+
 
     def add_jobs(self, k: int):
```

Comment:
In order to increase amount of jobs to send out should we add jobs until we can't sample enough miners anymore?